### PR TITLE
RFC: gpio: add configuration for DMA request

### PR DESCRIPTION
**Pull request by jfischer-no**
This patch adds GPIO configuration flags for DMA request.
The configured GPIO can be used to trigger a DMA transfer,
this is very useful for a parallel interface, so a gpio
can triggers the transfert from peripheral to memory or vice versa.

### CODE REVIEW

**Comment by pfalcon** [comment]
@jfischer-phytec-iot : The review you'll probably get for this is: "But what hardware supports that? Where're the drivers which support this feature?"

**Comment by tbursztyka** [comment]
I wonder how this is going to fit with existing DMA API. (And DMA API would need some rework to really fit all uses cases and be as generic as possible for all SoC btw).

**Comment by jfischer-phytec-iot** [comment]
>"But what hardware supports that? Where're the drivers which support this feature?"

e.g. NXP Kinetis µC. PR for the mcux shim DMA driver will follow soon.

>I wonder how this is going to fit with existing DMA API.

The DMA API is not the best, but I did not have to change anything. It only configures the trigger for the DMA transfer request.

**Comment by nashif** [comment]
> gpio: add configuration for DMA request
This patch adds GPIO configuration flags for DMA request.
The configured GPIO can be used to trigger a DMA transfer,
this is very useful for a parallel interface, so a gpio
can triggers the transfert from peripheral to memory or vice versa.


can trigger the transfer

File: include/gpio.h [hunk 1]
```diff
@@ -189,6 +189,20 @@ extern "C" {
...
+
+/** GPIO pin trigger DMA requenst on falling edge. */
+#define GPIO_DMA_REQ_EDGE_LOW		(1 << GPIO_DMA_REQ_POS)
+
+/** GPIO pin trigger DMA requenst on rising edge. */
+#define GPIO_DMA_REQ_EDGE_HIGH		(2 << GPIO_DMA_REQ_POS)
+
+/** GPIO pin trigger DMA requenst on rising and falling edge. */
+#define GPIO_DMA_REQ_EDGE_DOUBLE	(3 << GPIO_DMA_REQ_POS)
+
```
Comment (nashif) [inline include/gpio.h]: Are those generic enough to be in the API?

Comment (jfischer-phytec-iot) [inline include/gpio.h]: I know this possibility to trigger DMA request directly from the GPIO, only with Kinetis MCUs. The other make it a little cumbersome over a timer configured for Input Capture Mode, e.g. STM32 http://www.st.com/resource/en/application_note/dm00169730.pdf

I do not know all the architectures, but if it is only possible with Kinetis MCUs, then it makes no sense to include it in the API. @MaureenHelm what do you think?

Comment (MaureenHelm) [reply include/gpio.h]: It's also possible on LPC MCUs, so it does make sense to include in the API.

**Comment by nashif** [comment]
outdated, closing

**Pull request closed without merging**